components/automate-ui/src/app/pages/+compliance/shared/reporting/stats.service.ts

```diff
@@ -215,15 +217,20 @@ export class StatsService {
   /* TODO Auth team: Helper functions that are only not private because
      they have unit testing. Should we delete their tests and make
      them private? */
-  getControlStatus(control): string {
+  getControlStatus(control): ControlStatus {
     const statuses = control.results.map(r => r.status);
-    const waived = control.waiver_data && control.waiver_data.skipped_due_to_waiver;
+    const waived = this.checkIfWaived(control.waived_str);
     if (waived) { return 'waived'; }
     if (statuses.every(s => s === 'passed')) { return 'passed'; }
     if (statuses.every(s => s === 'skipped')) { return 'skipped'; }
     return 'failed';
   }
 
+  private checkIfWaived(waivedStatus: string): boolean {
+    return waivedStatus === 'yes_run' || waivedStatus === 'yes';
```

Comment:
Ultimately the error was coming from we needed to be checking against if `control.waived_str` is equal to `yes` or `yes_run`.  If it is, this means its waived.
`yes` means that the item was waived, but chosen to not run
`yes_run` means that the item was waived, but still run.